packages/aws-cdk/lib/api/hotswap/appsync-mapping-templates.ts

```diff
@@ -0,0 +1,100 @@
+import * as AWS from 'aws-sdk';
+import { ISDK } from '../aws-auth';
+import { EvaluateCloudFormationTemplate } from '../evaluate-cloudformation-template';
+import { ChangeHotswapImpact, ChangeHotswapResult, HotswapOperation, HotswappableChangeCandidate, lowerCaseFirstCharacter, transformObjectKeys } from './common';
+
+export async function isHotswappableAppSyncChange(
+  logicalId: string, change: HotswappableChangeCandidate, evaluateCfnTemplate: EvaluateCloudFormationTemplate,
+): Promise<ChangeHotswapResult> {
+  const isResolver = change.newValue.Type === 'AWS::AppSync::Resolver';
+  const isFunction = change.newValue.Type === 'AWS::AppSync::FunctionConfiguration';
+
+  if (!isResolver && !isFunction) {
+    return ChangeHotswapImpact.REQUIRES_FULL_DEPLOYMENT;
+  }
+
+  for (const updatedPropName in change.propertyUpdates) {
+    if (updatedPropName !== 'RequestMappingTemplate' && updatedPropName !== 'ResponseMappingTemplate') {
+      return ChangeHotswapImpact.REQUIRES_FULL_DEPLOYMENT;
+    }
+  }
+
+  const resource = change.newValue.Properties;
+  const evaluatedResource = {
+    ...await evaluateCfnTemplate.evaluateCfnExpression({
+      ...(resource ?? {}),
+    }),
+  };
+  const sdkCompatibleResource = transformObjectKeys(evaluatedResource, lowerCaseFirstCharacter);
+
+  const resourcePhysicalName = await evaluateCfnTemplate.establishResourcePhysicalName(logicalId, isFunction ? resource?.Name : null);
```

Comment:
So, on this one, I passed `null` because of the condition in `establishResourcePhysicalName`:

```
if (physicalNameInCfnTemplate != null) {...
```

I was too lazy to look up whether `undefined` will flow through that correctly.

Reply:
Do you know the answer?

Reply:
I do, and yes, `undefined` will work there 🙂.